### ISSUE TITLE
Signature Component

### ISSUE THREAD

**Issue by vikonava** [cerner/terra-core#692]
# Issue Description
Create a Signature Component.

## Issue Type

- [x] New Feature
- [ ] Enhancement
- [ ] Bug
- [ ] Other

## Expected Behavior
Should generate a canvas element that captures the user input when dragging mouse/finger on the device's screen. Line segments of the signature would be returned on a callback method along with the most recently drawn line.  Line Color and Line Width can be defined by the consumer.

**Comment by e3ujwh** [comment]
# Tech Design for terra-signature
## Summary:
The terra-signature component provides users a way to define a signature with a series of line segments.  The component can be define the line width, line color and optionally display a signature if the line segments have already been recorded.

## React Props:
| Prop        | Type | Default | Description  |
|----------|------|---------|--------------|
| lineWidth | number | 1    | Sets the line width |
| lineColor | string | #000000    | Sets the line color |
| lineSegments | array | undefined    | Line segments that define signature |
| onChange | func | undefined    | Callback function that returns all line segments and last drawn line segment |

## Example:
```
import React from 'react';

class SignatureExample extends React.Component {

  handleChange(event, lineSegments, lastSegment){
    console.log(JSON.stringify(lastSegment));
  }

  render() {
    return (      
      <div style={{ height: '100px', width: '100%', border: '1px solid black'}} >
        <Signature ref={instance => { this.signature = instance; }} 
               lineWidth={4} 
               lineColor={'#00008B'} 
               lineSegments={[{"x1":116,"y1":53,"x2":116,"y2":53}] 
               onChange={this.handleChange.bind(this)}/>
        <button onClick={() => {this.signature.clearSignature()}}>Clear</button>
      </div>
    );
  }
}
```

## CSS Classes:
| Selector         | Description |
|------------------|-------------|
| .terra-Signature | Used in canvas of signature component |

**Comment by dkasper-was-taken** [comment]
@neilpfeiffer  Do we want lineWidth and lineColor to be themeable via css-modules, or do we want to keep the params? And if we keep the params, do we need UX predefined values for available options?

**Comment by neilpfeiffer** [comment]
Lets reduce the props to just:
- lineWidth (enum of 4 sizes: I can provide list)
- lineSegments (array)
- onChange (func)

And we'll switch these to themeable variables:
- color
- background-color

**Comment by PaulSolDev** [comment]
Orion JIRA created

**Comment by bjankord** [comment]
+1 on tech design with addition of feedback from @neilpfeiffer.

**Comment by neilpfeiffer** [comment]
+1 -- API and initial implementation looks good to move forward with.

**Comment by bjankord** [comment]
This has been released as [terra-signature@0.1.0](https://www.npmjs.com/package/terra-signature)
